### PR TITLE
DC-1662 stride auth for adviser response

### PR DESCRIPTION
**Pull request by sammhmrc**
Adds stride auth to create response as well as auth util to ITs to make things easier in the future.

DO NOTMERGE UNTIL WE HAVE A PROPER AUTH-TEST VERISON

### CODE REVIEW

File: it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala [hunk 1]
```diff
@@ -66,14 +58,123 @@ class IntegrationTest extends WordSpec with Matchers with  CanCreateAuthority wi
...
+      val jsonString =
+        s"""
+           | {
+           |   "content": "$content"
+           | }
+    """.stripMargin
+
+      val validMessageId = MessageUtil.getValidMessageId()
+      val message = Json.parse(jsonString).as[JsObject]
+      val response = httpClient.url(resource(s"/two-way-message/message/advisor/$validMessageId/reply")).withHeaders(AuthUtil.buildStrideToken()).post(message).futureValue
```
Comment (andy1138) [inline it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala]: should this be   Await.result(httpClient.url(.....) .    (ie the same as used in buildStrideToken )

File: it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala [hunk 2]
```diff
@@ -66,14 +58,123 @@ class IntegrationTest extends WordSpec with Matchers with  CanCreateAuthority wi
...
+
+  object MessageUtil {
+    import play.api.libs.json.{Json, Reads}
+
+    implicit val deserialiser: Reads[MessageId] = Json.reads[MessageId]
+
+    case class MessageId(val id: String)
+
+    def getValidMessageId() = {
+      val content = UUID.randomUUID().toString
```
Comment (andy1138) [inline it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala]: should be a base64 string not at UUID, just use https://www.base64encode.org/ and create 'hello world' or something tjhen use the resulting string

File: it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala [hunk 3]
```diff
@@ -66,14 +58,123 @@ class IntegrationTest extends WordSpec with Matchers with  CanCreateAuthority wi
...
-      val response = httpClient.url(resource("/two-way-message/message/customer/0/submit")).withHeaders(authHeader).post(message).futureValue
+      val response = httpClient.url(resource("/two-way-message/message/customer/0/submit")).withHeaders(AuthUtil.buildUserToken()).post(message).futureValue
 
       response.status shouldBe 400
     }
   }
+
+  "Advisor responding" should {
+    "Forbidden when no access token" in {
+      val content = UUID.randomUUID().toString
```
Comment (andy1138) [inline it/uk/gov/hmrc/twowaymessage/IntegrationTest.scala]: see comment below

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -34,7 +34,7 @@ object AppDependencies {
 
   val test = Set(
     "uk.gov.hmrc" %% "hmrctest" % "3.2.0" % "test,it",
-    "uk.gov.hmrc" %% "auth-test" % "5.1.0" % "it",
+    "uk.gov.hmrc" %% "auth-test" % "999-SNAPSHOT" % "it",
```
[the remark this answers is not in the document]
Comment (sammhmrc) [reply project/AppDependencies.scala]: Aye. See the description 😉

**Pull request merged**